sympy/calculus/util.py

```diff
@@ -190,7 +193,33 @@ def function_range(f, symbol, domain):
                 else:
                     vals += FiniteSet(f.subs(symbol, limit_point))
 
-            solution = solveset(f.diff(symbol), symbol, interval)
+            # Replace symbol with dummy with proper assumptions if the symbol
+            # not already has it
+            l = (interval.inf, interval.sup)
+            if all(i.is_nonnegative for i in l) and not symbol.is_nonnegative:
+                d = Dummy(positive=True)
+            elif all(i.is_nonpositive for i in l) and not symbol.is_nonpositive:
```

Comment:
I think that this should be abstracted out to a function (even if incomplete and private) because it is something that can be useful in many situations.

Reply:
Yeah, I thought that there was some, but stole it directly from the integral code.

However, I seem to recall that there may be something in concrete that possibly can be modified (as it probably has integer constraints).

Reply:
cf #19523

Reply:
```python
>>> from sympy.core.assumptions import *
>>> l=Tuple(1, oo)
>>> symbol = Dummy(real=1)
>>> common = common_assumptions(l)
>>> if failing_assumptions(symbol, **common):
...    reps = {symbol: Dummy(**common)}
...
>>> failing_assumptions(reps[symbol], **common)
{}
>>> len(failing_assumptions(symbol, **common))
13
```